Alignment/OfflineValidation/python/TkAlAllInOneTool/Zmumu_cfg.py

```diff
@@ -0,0 +1,180 @@
+)
+
+###################################################################
+# Standard includes
+###################################################################
+process.load("FWCore.MessageService.MessageLogger_cfi")
+process.load("RecoVertex.BeamSpotProducer.BeamSpot_cff")
+process.load("Configuration.StandardSequences.Services_cff")
+process.load("Configuration.StandardSequences.GeometryRecoDB_cff")
+process.load("Configuration.StandardSequences.MagneticField_cff")
+process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+
+###################################################################
+# Messages
+###################################################################
+process.load("FWCore.MessageLogger.MessageLogger_cfi")
+process.MessageLogger.cerr.FwkReport.reportEvery = 1000
+
+###################################################################
+# Load and configure Track Refitter
+###################################################################
+process.load("RecoTracker.TrackProducer.TrackRefitters_cff")
+import RecoTracker.TrackProducer.TrackRefitters_cff
+process.TrackRefitter = process.TrackRefitterP5.clone(
+    src =  'ALCARECOTkAlZMuMu',
+    TrajectoryInEvent = True,
+    TTRHBuilder = "WithAngleAndTemplate",
+    NavigationSchool = "",
+)
+
+###################################################################
+# Global Tag
+###################################################################
+process.load("Configuration.StandardSequences.FrontierConditions_GlobalTag_cff")
+from Configuration.AlCa.GlobalTag import GlobalTag
+process.GlobalTag = GlobalTag(process.GlobalTag, config["alignment"].get("globaltag", "auto:run2_data"))
+
+###################################################################
+# Load conditions if wished
+###################################################################
+import CalibTracker.Configuration.Common.PoolDBESSource_cfi  
+if "conditions" in config["alignment"]:
+    from CalibTracker.Configuration.Common.PoolDBESSource_cfi import poolDBESSource
+    for condition in config["alignment"]["conditions"]:
+        setattr(process, "conditionsIn{}".format(condition), poolDBESSource.clone(
+            connect = cms.string(str(config["alignment"]["conditions"][condition]["connect"])),
+            toGet = cms.VPSet(cms.PSet(record = cms.string(str(condition)),
+                                       tag = cms.string(str(config["alignment"]["conditions"][condition]["tag"]))
+                                      )
+                              )
+            )
+        )
+
+        setattr(process, "prefer_conditionsIn{}".format(condition), cms.ESPrefer("PoolDBESSource", "conditionsIn{}".format(condition)))
+
+###################################################################
+# The Di Muon Mass Validation module
+###################################################################
+process.DiMuonMassValidation = cms.EDAnalyzer('DiMuonValidation',

```

Comment:
In general, here, it would be worthwhile to give the possibility to the lambda user to override all the following options from the config file. The arguments shouldn't be mandatory, so if the value exists, use it, otherwise use the default value.

Reply:
I do not quite understand this comment. Do you mean change those values to something like this:
input = cms.untracked.int32(config["validation"].get("maxevents", 2000000)) ?

Reply:
That's the idea: don't hardcode the value, but give the possibility to the aligner to overwrite these values from the config file (while keeping a default value if he does not want to overwrite it). Take a look at the implementation of other validations.

Reply:
These are physical geometrical boundaries. Was the Z->mm validation ever run with partial coverage of the phase-space ?
In my opinion only the number of bins could make sense to be configurable. Too much freedom to the users quickly leads to nonsense.

Reply:
In the past we did it.

Anyway, the smart aligner will know where to change the value if necessary. I agree that it is an exotic case, so ok if you prefer fixing it once for all.

However, the number of bins is certainly sth needed indeed.

Reply:
> In the past we did it.

I think what was done in the past was to limit the acceptance of the single muon track to disentangle barrel from endcaps, not to plot the mass bias only up to some  η or φ value. The two things are different.
By the way, the current code doesn't provide for the former functionality.